src/frontend/screens/Settings/ProjectSettings/SyncActionSheet.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import {BottomSheetModalMethods} from '@gorhom/bottom-sheet/lib/typescript/types';
+import {defineMessages, useIntl} from 'react-intl';
+import Warning from '../../../images/Warning.svg';
+import {
+  BottomSheetModal,
+  BottomSheetModalContent,
+} from '../../../sharedComponents/BottomSheetModal';
+
+const m = defineMessages({
+  cancel: {
+    id: 'screens.SyncActionSheet.cancel',
+    defaultMessage: 'Cancel',
+  },
+});
+
+type SyncActionSheetProps = {
+  title: string;
+  description: string;
+  confirmActionText: string;
+  confirmAction: () => void;
+  isOpen: boolean;
+  onDismiss: () => void;
+};
+
+export const SyncActionSheet = React.forwardRef<
+  BottomSheetModalMethods,
+  SyncActionSheetProps
+>(
+  (
+    {title, description, confirmActionText, confirmAction, isOpen, onDismiss},
+    sheetRef,
+  ) => {
+    const {formatMessage: t} = useIntl();
+
+    return (
+      <BottomSheetModal isOpen={isOpen} ref={sheetRef} onDismiss={onDismiss}>
+        <BottomSheetModalContent
+          title={title}
+          description={description}
+          icon={<Warning width={60} height={60} />}
+          buttonConfigs={[
+            {
+              text: t(m.cancel),
+              onPress: onDismiss,
+              variation: 'outlined',
+            },
+            {
+              text: confirmActionText,
+              variation: 'filled',
+              onPress: confirmAction,
+            },
+          ]}
+        />
+      </BottomSheetModal>
+    );
+  },
+);
```

Comment:
suggestion: it's not clearly documented but it's usually unnecessary to create components that wrap the `BottomSheetModal` component as it's designed to be co-located with the usage of the `useBottomSheetModal()` hook. Instead, define what's needed for the `BottomSheetModalContent`:

```suggestion
type SyncActionSheetProps = {
  title: string;
  description: string;
  confirmActionText: string;
  confirmAction: () => void;
  onDismiss: () => void;
};

export const SyncActionSheet = ({
  title,
  description,
  confirmActionText,
  confirmAction,
  onDismiss,
}: SyncActionSheetProps) => {
  const {formatMessage: t} = useIntl();

  return (
    <BottomSheetModalContent
      title={title}
      description={description}
      icon={<Warning width={60} height={60} />}
      buttonConfigs={[
        {
          text: t(m.cancel),
          onPress: onDismiss,
          variation: 'outlined',
        },
        {
          text: confirmActionText,
          variation: 'filled',
          onPress: confirmAction,
        },
      ]}
    />
  );
};
```

Might also be worth renaming the component to make it clearer that it's just the content of the bottom sheet.

Then in `SyncSettings`, update the usage to something like the following:

```tsx
<BottomSheetModal ref={sheetRef} isOpen={modalState.isOpen}>
  <SyncActionSheet
    title={
      modalState.type === 'previews'
        ? t(m.syncPreviewsBottomSheet)
        : t(m.syncEverythingBottomSheet)
    }
    description={
      modalState.type === 'previews'
        ? t(m.syncPreviewsDescriptionBottomSheet)
        : t(m.syncEverythingDescriptionBottomSheet)
    }
    confirmActionText={
      modalState.type === 'previews'
        ? t(m.syncPreviewsBottomSheetConfirm)
        : t(m.syncEverything)
    }
    confirmAction={handleConfirm}
    onDismiss={() => {
      setModalState({type: null, isOpen: false});
      closeSheet();
    }}
  />
</BottomSheetModal>
```

Reply:
Ok done.